agent/consul/leader_connect_test.go

```diff
@@ -420,8 +420,13 @@ func TestCAManager_RenewIntermediate_Vault_Primary(t *testing.T) {
 		intermediatePEM = newIntermediatePEM
 	})
 
-	_, activeRoot, err = store.CARootActive(nil)
+	codec := rpcClient(t, s1)
+	roots := structs.IndexedCARoots{}
+	err = msgpackrpc.CallWithCodec(codec, "ConnectCA.Roots", &structs.DCSpecificRequest{}, &roots)
```

Comment:
I changed from a direct state store call to an RPC call because we need the `TrustDomain` below to create a valid spiffe ID.

I'm attempting to move where we store this value (see #11687) so I prefer fetching it from a public interface instead of reaching into the state store config.